src/core/etl/src/Flow/ETL/DSL/functions.php

```diff
@@ -593,9 +595,9 @@ function concat(ScalarFunction ...$functions) : Concat
     return new Concat(...$functions);
 }
 
-function hash(ScalarFunction $function, string $algorithm = 'xxh128', bool $binary = false, array $options = []) : Hash
+function hash(ScalarFunction $function, Algorithm $algorithm = new NativePHPHash()) : Hash
```

Comment:
IMO this should be noted in BC breaks.

Reply:
I highly doubt anyone even used that function with any other algorithm, until we release `1.0` I would prefer to put only truly problematic migrations to UPGRADE.md to not pollute that file. This one is super straightforward and PHP linter should fail so I don't think there is even a risk that someone will accidentally push it to the production